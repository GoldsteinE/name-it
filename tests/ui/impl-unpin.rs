use name_it::name_it;

#[name_it(Foo)]
async fn foo() {
}

impl Unpin for Foo<'_> {}

fn assert_unpin(_: impl Unpin) {}

fn main() {
    assert_unpin(foo());
}
