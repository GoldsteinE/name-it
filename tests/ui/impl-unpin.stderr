error[E0117]: only traits defined in the current crate can be implemented for types defined outside of the crate
 --> tests/ui/impl-unpin.rs:7:1
  |
7 | impl Unpin for Foo<'_> {}
  | ^^^^^^^^^^^^^^^-------
  | |              |
  | |              `Named` is not defined in the current crate
  | impl doesn't use only types from inside the current crate
  |
  = note: define and implement a trait or new type instead
